tribe-ext-schedule-day-view.php

```diff
@@ -0,0 +1,886 @@
+ * License:         GPL version 3 or any later version
+ * License URI:     https://www.gnu.org/licenses/gpl-3.0.html
+ * Text Domain:     tribe-ext-schedule-day-view
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Schedule_Day_View' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Schedule_Day_View extends Tribe__Extension {
+
+		/**
+		 * The prefix used for settings or whatever else needs to be namespaced.
+		 */
+		const PREFIX = 'tribe_ext_sch_day_view';
+
+		/**
+		 * Time slot name used within each WP_Post.
+		 *
+		 * @see Tribe__Extension__Schedule_Day_View::PREFIX
+		 *
+		 * @return string
+		 */
+		public function time_slot_name() {
+			return self::PREFIX . '_time_slot';
+		}
+
+		/**
+		 * The current time slot's array of arguments. Used in the loop template
+		 * file per time slot.
+		 *
+		 * @var array
+		 */
+		public $current_time_slot_args = array();
+
+		/**
+		 * The list of The Events Calendar's template files to override with
+		 * which of this plugin's template files.
+		 *
+		 * @return array
+		 */
+		private function templates() {
+			return array(
+				'day/single.php'          => 'src/views/day/single.php',
+				'day/single-event.php'    => 'src/views/day/single.php',
+				'day/single-featured.php' => 'src/views/day/single.php',
+				'day/loop.php'            => 'src/views/day/loop.php',
+			);
+		}
+
+		/**
+		 * Set the minimum required version of The Events Calendar
+		 * and this extension's URL.
+		 */
+		public function construct() {
+			// Tribe__Assets::maybe_get_min_file() requires v4.3
+			$this->add_required_plugin( 'Tribe__Events__Main', '4.3' );
+			$this->set_url( 'https://theeventscalendar.com/extensions/schedule-day-view/' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-schedule-day-view', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			/**
+			 * Protect against fatals by specifying the required minimum PHP
+			 * version. Make sure to match the readme.txt header.
+			 *
+			 * @link https://secure.php.net/manual/en/migration54.new-features.php
+			 * 5.4: Traits, Short Array Syntax, and $this within Closures
+			 */
+			$php_required_version = '5.4';
```

Comment:
I see uses of 5.3 syntax (anonymous functions for example) but I was curious what we're using that is only available with 5.4+?

Reply:
`$this` within the anonymous function

I could change it to be 5.3 if you want...

Reply:
Per Slack Barry said to leave it as-is.